src/behave_tests/features/environment.py

```diff
@@ -0,0 +1,99 @@
+
+import os
+
+from behave import fixture
+from jinja2 import Template
+
+KCLI_PLANS_DIR = "kcli_plans"
+
+KCLI_PLAN_TEMPLATE = """
+{% for number in range(0, nodes) %} 
+{{ prefix }}-node-0{{ number }}:
+ image: {{ image }}
+ numcpus: {{ numcpus }}
+ memory: {{ memory }}
+ reserveip: true
+ reservedns: true
+ sharedkey: true
+ domain: {{ domain }}
+ nets:
+  - {{ network }}
+ disks: {{ disks }}
+ pool: {{ pool }}
+ {% if ceph_dev_folder is defined %}
+ sharedfolders: [{{ ceph_dev_folder }}]
+ {% endif %}
+ cmds:
+ - yum -y install python3 chrony lvm2 podman
+ - sed -i "s/SELINUX=enforcing/SELINUX=permissive/" /etc/selinux/config
+ - echo "after installing the python3"
+ - setenforce 0
+ {% if number == 0 %}
+ scripts:
+  - bootstrap-cluster_dev.sh
+ {% endif %}
+{% endfor %}
+"""
+
+Configuration = {
+    "nodes": 1,
+    "pool": "default",
+    "network" : "default",
+    "domain" : "cephlab.com",
+    "prefix" : "ceph",
+    "numcpus" : 1,
+    "memory" : 1024,
+    "image" : "fedora33",
+    "notify" : False,
+    "admin_password": "password",
+    "disks": [150, 3]
+}
+
+
+def _parse_memory_value(value):
+    if value.endswith("gb"):
+        value = value.replace('gb', '')
+        return int(value) * 1024
+    elif value.endswith("mb"):
+        return value.replace('mb', '')
+
+
+def _filter_feature_spec(feature):
+    feature = feature.replace('.', '')
+    feature = feature.replace(',', '')
+    if feature.endswith("gb") or feature.endswith("mb"):
+        feature = _parse_memory_value(feature)
+    return feature
+
+def _parse_feature_specifications(feature_specs):
+    specs = [
+         _filter_feature_spec(spec)
+         for spec in feature_specs.lower().split()
+        ]
+    print(f"specs : \n{specs}")
+    Configuration['nodes'] = int(specs[specs.index('nodes')-1])
+    Configuration['cpus'] = int(specs[specs.index('cpus') -1])
+    Configuration['memory'] = specs[specs.index('ram') -1 ]
+    Configuration['disk'] = [
+        specs[specs.index('each')-1],
+        specs[specs.index('storage') -1],
```

Comment:
This can be heavily improved using [regular expressions](https://docs.python.org/3/library/re.html ):

example:
```
>>> m = re.match(r"(?P<first_name>\w+) (?P<last_name>\w+)", "Malcolm Reynolds")
>>> m.group('first_name')
'Malcolm'
>>> m.group('last_name')
'Reynolds'
```

In our case, we are trying to extract information from a couple of lines with fixed structure like:
```
          - 3 nodes with 4Gb RAM, 2 CPUs, and 3 storage devices of 20Gb each.
          - Using Fedora32 image in each node
```

So it must no be difficult to extract directly the information we need to the configuration dict